### PR TITLE
removing ledger backend module

### PR DESCRIPTION
**Pull request by gaboraranyossy-da**
Removed the unnecessary `ledger-backend` module/abstraction.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/SandboxIndexService.scala [hunk 1]
```diff
@@ -235,28 +151,27 @@ class SandboxIndexService(
...
+                      case (offset, _) =>
+                        //note that we can have gaps in the increasing offsets!
+                        (offset + 1) < end.toLong //api offsets are +1 compared to backend offsets
                     },
                     inclusive = true // we need this to be inclusive otherwise the stream will be hanging until a new element from upstream arrives
                   )
-                  .filter(_.offset.toLong < end.toLong)
+                  .filter(_._1 < end.toLong)
             }
             // we MUST do the offset comparison BEFORE collecting only the accepted transactions,
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/SandboxIndexService.scala]: Please update the comment with the changes (e.g. the concept of "accepted transaction" doesn't exist anymore).

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/SandboxIndexService.scala [hunk 2]
```diff
@@ -286,53 +213,30 @@ class SandboxIndexService(
...
-                _,
-                recordTime,
-                offset,
-                _,
-                _,
-                Some(appId),
-                Some(commandId))
-              if (appId == applicationId.unwrap && parties.contains(submitter)) =>
+        ledger.ledgerEntries(Some(absBegin.toLong)).collect {
+          case (offset, t: LedgerEntry.Transaction)
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/SandboxIndexService.scala]: Hm, not sure but before we were collecting transaction only when they had an `appId` and `commandId` (see line 300 and 301). We don't do that anymore and it even seems that `LedgerEntry.Transaction` always has an application Id, submitting party, and command Id. I think we might need to change that if we want to reuse it for the IndexService+ReadService implementation (where we might not always have those fields).

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/TransactionConversion.scala [hunk 3]
```diff
@@ -23,7 +22,8 @@ import scala.collection.breakOut
 trait TransactionConversion {
 
   def acceptedToDomainFlat(
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/TransactionConversion.scala]: Please rename from `acceptedToDomainFlat` to something like `ledgerEntryTxToDomainFlat`.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/TransactionConversion.scala [hunk 4]
```diff
@@ -40,25 +40,25 @@ trait TransactionConversion {
...
+          trans.recordedAt,
           filteredEvents,
-          domain.LedgerOffset.Absolute(trans.offset),
+          offset,
           None
         ))
     } else None
   }
 
   def acceptedToDomainTree(
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/TransactionConversion.scala]: Please change the name.

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/SubmitterInfo.scala [hunk 5]
```diff
@@ -30,5 +30,5 @@ final case class SubmitterInfo(
     submitter: Party,
     applicationId: ApplicationId,
     commandId: CommandId,
-    maxRecordTime: Timestamp,
+    maxRecordTime: Timestamp, //TODO: this should be a regular Instant
```
Comment (garyverhaegen-da) [inline ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/SubmitterInfo.scala]:
```suggestion
    maxRecordTime: Timestamp //TODO: this should be a regular Instant
```

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/SandboxIndexService.scala (hunk 2 above)
Comment (gaboraranyossy-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/SandboxIndexService.scala]: good point, but I'd consider doing that only when approaching our 2. milestone. I would start that with a gap analysis and lay out a plan of actions upfront.

**Comment by gaboraranyossy-da** [comment]
@garyverhaegen-da @gerolf-da your comments have been addressed

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/SandboxIndexService.scala (hunk 2 above)
Comment (gerolf-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/SandboxIndexService.scala]: agreed 👍

**Pull request merged**